### PR TITLE
User validations

### PR DESCRIPTION
**Pull request by rae-stanton**
Adds:
Validates that name, email, and password are present when creating a new user with messages
Adds bcrypt for password encryption
Adds comparison with existing user that email or name are not already in use (with message).

To do:
Need to add tests for user validations
Need to add specifications for user creation (length, characters, etc) for correct user 
Wire up form to accept input from users 
Write tests for form on signup

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -1,4 +1,11 @@
...
-  def new
+  def create
+    user = User.create()
   end
+
+  private
+  def user_params
+    params.permit(:username, :password)
+  end
+
```
Comment (spaulsandhu) [inline app/controllers/users_controller.rb]: I don't think this controller does anything right now, where are we hooking into it?

File: db/migrate/20230214205506_create_users.rb [hunk 2]
```diff
@@ -2,7 +2,8 @@ class CreateUsers < ActiveRecord::Migration[7.0]
   def change
     create_table :users do |t|
       t.string :name
-      t.string :email
+      t.string :email, null: false
+      t.string :password_digest
```
Comment (spaulsandhu) [inline db/migrate/20230214205506_create_users.rb:6]: *Never, ever* edit an existing migration that has been put into production. It's relatively safe to edit a migration that has not been merged into `main` (for example, while you are working on a feature branch on your own). But once a migration has made it into `main` and is on other people's machines, consider it a historical record that should never be changed.

Please see this snippet from the official Rails guide here: https://guides.rubyonrails.org/v7.0/active_record_migrations.html#changing-existing-migrations

Any time you need to modify an existing table, your best practice is to generate a generic migration that you manually edit to modify the table as you need.

File: db/seeds.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
...
-#
-#   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
-#   Character.create(name: "Luke", movie: movies.first)
+User.create!(name: "Faust Lune",
+             email: "flune@henlo.com",
+             password: "foobar",
+             password_confirmation: "foobar",
+             admin: true,
+             activated: true,
+             activated_at: Time.zone.now)
```
Comment (spaulsandhu) [inline db/seeds.rb]: Missing extra new line.

File: spec/views/signup.html.erb_spec.rb [hunk 4]
```diff
@@ -0,0 +1,10 @@
+require "rails_helper"
+
+RSpec.describe StaticPagesController, type: :controller do
+  describe "GET /signup" do
+  it "shows a form" do
+    get :signup
+      expect(response).to render_template "static_pages/signup"
+    end
+  end
+end
```
Comment (spaulsandhu) [inline spec/views/signup.html.erb_spec.rb]: You're missing an extra newline here too.

**Review by spaulsandhu — CHANGES_REQUESTED** [review]
You're making some great progress but I think we need to clean some things up before we can merge this down. Please see my comments about the migrations best practices. 

We can work on a few lessons together this weekend to go over some of the finer details of things you can do fix some of the issues with this PR.

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (rae-stanton) [reply app/controllers/users_controller.rb]: It doesn't do anything yet, I'm currently working on wiring up the signup to this

**Comment by rae-stanton** [comment]
This should be the remainder of the changes for user validations now that the rspec is finished with what is currently going on and form submits properly with tests. LMK if there are any other changes needed before merge.

**Review by spaulsandhu — APPROVED** [review]

**Pull request merged**